ci-operator/step-registry/baremetalds/e2e/upgrade/baremetalds-e2e-upgrade-workflow.yaml

```diff
@@ -0,0 +1,24 @@
+workflow:
+  as: baremetalds-e2e-upgrade
+  steps:
+    allow_skip_on_success: true 
+    pre:
+    - ref: baremetalds-packet-setup
+    - ref: ipi-install-rbac
+    - ref: baremetalds-devscripts-setup
+    test:
+      - ref: baremetalds-e2e-conf
+      - ref: baremetalds-e2e-test
+      - ref: baremetalds-e2e-upgrade-upgrade
```

Comment:
Splitting the upgrade step from the test step will cause an unnecessary tests execution during the first `e2e-test` step (which is something already exercised in the standard workflow), thus increasing also the overall time for the job. It would be preferred to merge the two steps together with a conditional execution (in a way similar to the other profiles) here: https://github.com/openshift/release/blob/8cb0233ac6d7a0a4c3b445fae410610ecd7d72b3/ci-operator/step-registry/baremetalds/e2e/test/baremetalds-e2e-test-commands.sh#L41